### ISSUE TITLE
🟣 `stellar-docs-code-snippet.md` claimable-balance-python

### ISSUE THREAD

**Issue by apaldiwal** [tyvdh/stellar-quest-bounties#73]
### Link the bounty file

https://github.com/stellar/new-docs/pull/472

### Mark your progress

- [x] 🔵&nbsp; Started working
- [ ] 🟢&nbsp; Ready for review
- [ ] 🟣&nbsp; Review completed

**Comment by apaldiwal** [comment]
I'm not sure if I categorized the code correctly, but I think it's equivalent to 6 unique function code blocks. What do you think? @tyvdh

**Comment by apaldiwal** [comment]
I'm not sure how to categorize the code into unique function blocks. What do you think?

**Comment by tyvdh** [comment]
Here's how it's currently being done.
https://github.com/tyvdh/stellar-quest-bounties/issues/68

**Comment by tyvdh** [comment]
Oh looks like you had an Issue here: https://github.com/tyvdh/stellar-quest-bounties/issues/49. Probably best to either re-open that one or else copy that data here and just begin to comment as PRs to the docs repo gets merged. Were they all merged?

**Comment by apaldiwal** [comment]
> Oh looks like you had an Issue here: #49. Probably best to either re-open that one or else copy that data here and just begin to comment as PRs to the docs repo gets merged. Were they all merged?

No, just this one. I can have a new column in the table where I can report the status of each PR. Does that work?

**Comment by tyvdh** [comment]
Yeah probably 🤔 No need to open a new issue for each merge I don't think. Just need to track the progress as it happens in the one Issue. It's a bit of a unique bounty but worth thinking about how best to keep track with it. Probably should still mark as 🟣 on that as the review is technically done or being done and as far as this bounty repo is concerned all that's left is to get you paid!

**Comment by apaldiwal** [comment]
> Yeah probably 🤔 No need to open a new issue for each merge I don't think. Just need to track the progress as it happens in the one Issue. It's a bit of a unique bounty but worth thinking about how best to keep track with it. Probably should still mark as 🟣 on that as the review is technically done or being done and as far as this bounty repo is concerned all that's left is to get you paid!

Done! I know you explained it earlier, but I'm still a little confused about how to count the unique code blocks in each example. It's a bit of a gray area really because there's a lot of ambiguity in how you split the code into functional blocks.

**Comment by apaldiwal** [comment]
> I agree 100%. Would love to hear your thoughts around this. Is per file better than per block? What do you think it's worth? I'd really love to ensure I'm being generous and fair while also being somewhat algorithmic/predictable about it.

I feel 'per file' is a better option for code involving operations because you can't just pick and choose a block of code from the example and submit a PR in the repo. You have to fully incorporate the example in code while ensuring syntactical consistency with the existing examples. For example, https://developers.stellar.org/docs/glossary/claimable-balance/

On the other hand, for code involving API Reference we can use 'per block' to award bounties because we only have to deal with one functional aspect at a time. For example, https://developers.stellar.org/api/aggregations/fee-stats/single/

Honestly, I am really indecisive when it comes to picking numbers for a bounty. So, I'm okay with whatever you end up deciding. :)

**Comment by ralphilius** [comment]
> Yeah probably 🤔 No need to open a new issue for each merge I don't think. Just need to track the progress as it happens in the one Issue. It's a bit of a unique bounty but worth thinking about how best to keep track with it. Probably should still mark as 🟣 on that as the review is technically done or being done and as far as this bounty repo is concerned all that's left is to get you paid!

Maybe we could link from the original PR so that we can have PR status in the review issue? 

I think tThis type of bounty is typical if we want some organization to sponsor a bounty to maintain their documentation as well.

**Comment by tyvdh** [comment]
> > Yeah probably 🤔 No need to open a new issue for each merge I don't think. Just need to track the progress as it happens in the one Issue. It's a bit of a unique bounty but worth thinking about how best to keep track with it. Probably should still mark as 🟣 on that as the review is technically done or being done and as far as this bounty repo is concerned all that's left is to get you paid!
> 
> Maybe we could link from the original PR so that we can have PR status in the review issue (e.g. #68) ?
> 
> I think tThis type of bounty is typical if we want some organization to sponsor a bounty to maintain their documentation as well.

Has he not done that?
![Screen Shot 2021-08-05 at 9 31 51 AM](https://user-images.githubusercontent.com/4383610/128358494-e9bf3d77-8d46-4d4d-a3ea-41bd4f4a1fab.png)

Or are you referring to something else?

**Comment by tyvdh** [comment]
@apaldiwal I just updated your original tables with my proposed award amounts. Do they seem fair? No is a correct answer, I want this to generously reflect the work you put in such to properly incentivize future work along these lines so think for others and not just for yourself.

If anyone else would like to chime in here feel free.
cc: @rahimklaber @ralphilius @Kanaye @Smephite @ElliotFriend @hanseartic

**Comment by ElliotFriend** [comment]
Why the difference between the two? It's been hard to keep up with all the discussions, so it's possible (rather, likely) that I've missed something.

At first glance 50 seems a bit low, but I have no real reasoning behind that, just kind of a "gut feeling."

**Comment by rahimklaber** [comment]
I think the rewards are fair. To me it seems that part of the problem is making it fair for people of multiple experience levels. If you are experienced the rewards will probably a lot for you, since you will be able to write examples quickly. But if you are newer it will take you more time. I guess this is fair though,  experienced people should be paid more.

For now I think 50 xlm per block is fair.

**Comment by tyvdh** [comment]
> Why the difference between the two? It's been hard to keep up with all the discussions, so it's possible (rather, likely) that I've missed something.

Good question. If you look at the code blocks for the code being committed I see three distinct blocks of code in the changed file. Undoubtedly the first block is much more complex than others but I see a pretty distinct collection of blocks. So that was my logic.

It's a tricky balance between fair and algorithmic so folks have some sense of what to expect and it's not up to how generous I'm feeling that day.

We can certainly bump to 100 XLM per code block as this is a particularly beneficial value add for our docs and does have a cap in that we only need Python examples once per block not everyone on planet earth can write more Python code snippet examples. So generous but limited.

Seem fair?

**Comment by ElliotFriend** [comment]
> Good question. If you look at the code blocks for the code being committed I see three distinct blocks of code in the changed file. Undoubtedly the first block is much more complex than others but I see a pretty distinct collection of blocks. So that was my logic.

You mean to tell me I'm expected to actually *look* at something in its **entirety** before commenting???!!! Sir! This is the internet! That's not how it works around here.

(Hopefully the profuse sarcasm has come across well, and I don't just look like a giant asshole XD)

That makes perfect sense! It all sounds good to me 👍

**Comment by ralphilius** [comment]
> > > Yeah probably 🤔 No need to open a new issue for each merge I don't think. Just need to track the progress as it happens in the one Issue. It's a bit of a unique bounty but worth thinking about how best to keep track with it. Probably should still mark as 🟣 on that as the review is technically done or being done and as far as this bounty repo is concerned all that's left is to get you paid!
> > 
> > 
> > Maybe we could link from the original PR so that we can have PR status in the review issue (e.g. #68) ?
> > I think tThis type of bounty is typical if we want some organization to sponsor a bounty to maintain their documentation as well.
> 
> Has he not done that?
> ![Screen Shot 2021-08-05 at 9 31 51 AM](https://user-images.githubusercontent.com/4383610/128358494-e9bf3d77-8d46-4d4d-a3ea-41bd4f4a1fab.png)
> 
> Or are you referring to something else?

Something like this one
![image](https://user-images.githubusercontent.com/4253551/128380144-3f4ba672-b77e-46d1-8e99-1dbc088c57ed.png)

I didn't know about hovering card, so that should be fine as well.

**Comment by tyvdh** [comment]
https://github.com/stellar/new-docs/pull/472
3b5481f434dee93cc1ebbe836623d7dd2147b37f9a578f569ea056d83aefc175

**Comment by ElliotFriend** [comment]
Changing this back to the `review` state, since we're waiting for merges on the remaining snippets.
